frontend/src/components/Permissions.tsx

```diff
@@ -1,274 +1,235 @@
 import { PlusCircle } from "lucide-react";
-import React, { useEffect, useState } from "react";
+import React from "react";
 import BudgetAmountSelect from "src/components/BudgetAmountSelect";
 import BudgetRenewalSelect from "src/components/BudgetRenewalSelect";
+import ExpirySelect from "src/components/ExpirySelect";
+import Scopes from "src/components/Scopes";
 import { Button } from "src/components/ui/button";
-import { Checkbox } from "src/components/ui/checkbox";
-import { Label } from "src/components/ui/label";
-import { useCapabilities } from "src/hooks/useCapabilities";
 import { cn } from "src/lib/utils";
 import {
   AppPermissions,
   BudgetRenewalType,
+  NIP_47_PAY_INVOICE_METHOD,
   Scope,
-  expiryOptions,
+  WalletCapabilities,
   iconMap,
   scopeDescriptions,
 } from "src/types";
 
 interface PermissionsProps {
+  capabilities: WalletCapabilities;
   initialPermissions: AppPermissions;
   onPermissionsChange: (permissions: AppPermissions) => void;
+  canEditPermissions?: boolean;
   budgetUsage?: number;
-  canEditPermissions: boolean;
   isNewConnection?: boolean;
 }
 
 const Permissions: React.FC<PermissionsProps> = ({
+  capabilities,
   initialPermissions,
   onPermissionsChange,
   canEditPermissions,
   isNewConnection,
   budgetUsage,
 }) => {
   const [permissions, setPermissions] = React.useState(initialPermissions);
-  const [days, setDays] = useState(isNewConnection ? 0 : -1);
-  const [expireOptions, setExpireOptions] = useState(!isNewConnection);
-  const { data: capabilities } = useCapabilities();
 
-  useEffect(() => {
-    setPermissions(initialPermissions);
-  }, [initialPermissions]);
+  const [isScopesEditable, setScopesEditable] = React.useState(
+    isNewConnection ? !initialPermissions.scopes.size : canEditPermissions
+  );
+  const [isBudgetAmountEditable, setBudgetAmountEditable] = React.useState(
+    isNewConnection
+      ? Number.isNaN(initialPermissions.maxAmount)
+      : canEditPermissions
+  );
+  const [isExpiryEditable, setExpiryEditable] = React.useState(
+    isNewConnection ? !initialPermissions.expiresAt : canEditPermissions
+  );
```

Comment:
Current flow (given nothing is passed in parameters)
permissions are set in NewApp as

`scopes: []`
`maxAmount:NaN` (as `AppPermissions` type expects a number)
`expiresAt: undefined` (thankfully it expects `Date | undefined`)

And hence we use these checks to identify if the params had any permission data

Reply:
Using NaN seems a bit strange, I'll take a look and see if there's another way